File: Wrapping/WrapITK/Java/Tests/notYetUsable/DicomSliceRead.java
/**
 * Example on the use of DicomImageIO for reading a single DICOM slice, rescale
 * the intensities and save it in a different file format.
 *
 */

import InsightToolkit.*;

public class DicomSliceRead
{
  public static void main( String argv[] )
  {
    System.out.println("DicomSliceRead Example");

    itkImageFileReaderUS2_Pointer reader = itkImageFileReaderUS2.itkImageFileReaderUS2_New();
    itkImageFileWriterUC2_Pointer writer = itkImageFileWriterUC2.itkImageFileWriterUC2_New();

    itkRescaleIntensityImageFilterUS2UC2_Pointer filter = itkRescaleIntensityImageFilterUS2UC2.itkRescaleIntensityImageFilterUS2UC2_New();

    filter.SetInput( reader.GetOutput() );
    writer.SetInput( filter.GetOutput() );

    itkDicomImageIO_Pointer dicomIO = itkDicomImageIO.itkDicomImageIO_New();
    
    reader.SetImageIO( dicomIO.GetPointer() );
    
    filter.SetOutputMinimum( (short)0 );
    filter.SetOutputMaximum( (short) 255);

    reader.SetFileName( argv[0] );
    writer.SetFileName( argv[1] );

    writer.Update();
  }

}

File: Wrapping/WrapITK/Python/Tests/notYetUsable/DicomSliceRead.py
#
#  Example on the use of DicomImageIO for reading a single DICOM slice, rescale
#  the intensities and save it in a different file format.
#

from InsightToolkit import *

from sys import argv

#
# Reads an image in  16bits/pixel 
# and save it as      8bits/pixel
#
reader = itkImageFileReaderUS2_New()
writer = itkImageFileWriterUC2_New()

dicomIO = itkDicomImageIO_New()

reader.SetImageIO( dicomIO.GetPointer() )

filter  = itkRescaleIntensityImageFilterUS2UC2_New()

filter.SetOutputMinimum( 0 )
filter.SetOutputMaximum(255)

filter.SetInput( reader.GetOutput() )
writer.SetInput( filter.GetOutput() )

reader.SetFileName( argv[1] )
writer.SetFileName( argv[2] )

writer.Update()


